mikupad.html

```diff
@@ -2157,13 +2344,228 @@
 	const [stoppingStrings, setStoppingStrings] = useSessionState('stoppingStrings', defaultPresets.stoppingStrings);
 	const [stoppingStringsError, setStoppingStringsError] = useState(undefined);
 	const [savedScrollTop, setSavedScrollTop] = useSessionState('scrollTop', defaultPresets.scrollTop);
-
+	const [modalState, setModalState] = useState({});
+	const [logitBias, setLogitBias] = useSessionState('logitBias', defaultPresets.logitBias);
+	const [logitBiasParam, setLogitBiasParam] = useState({});
+	const [logitBiasTemp, setLogitBiasTemp] = useState([]);
+	const [logitBiasSorted, setLogitBiasSorted] = useState([]);
+	const [logitBiasInput, setLogitBiasInput] = useState({power:"0",string:""});
 	const [contextLength, setContextLength] = useSessionState('contextLength', defaultPresets.contextLength);
 	const [memoryTokens, setMemoryTokens] = useSessionState('memoryTokens', defaultPresets.memoryTokens);
-
 	const [authorNoteTokens, setAuthorNoteTokens] = useSessionState('authorNoteTokens', defaultPresets.authorNoteTokens);
 	const [authorNoteDepth, setAuthorNoteDepth] = useSessionState('authorNoteDepth', defaultPresets.authorNoteDepth);
 
+	const handleLogitBiasInput = (key,value) => {
+		setLogitBiasInput((prevLogitBiasInput) => {
+			return {
+				...prevLogitBiasInput,
+				[key]: value
+			}
+		});
+
+	}
+
+	const logitBiasAdd = async (biasPower="",biasString="",origValue="") => {
+		setLastError(undefined)
+		// abort if no input or power is NaN
+		if(!biasString) {return}
+		if (isNaN(+biasPower) || biasPower == "") { 
+			setLastError("Error: Bias is NaN")

```

Comment:
`setLastError` sets the prediction error, it would be better to have a specific state for the logit bias errors.
![image](https://github.com/lmg-anon/mikupad/assets/139719567/a0ae4843-32bd-4891-ae89-cd796dafa999)